src/modules/emotes/socket.js

```diff
@@ -0,0 +1,66 @@
+import socketClient, {EventNames} from '../../socket-client.js';
+import {getCurrentChannel} from '../../utils/channel.js';
+import twitch from '../../utils/twitch.js';
+import watcher from '../../watcher.js';
+import channelEmotes from './channel-emotes.js';
+
+function validChannelDestination(channel) {
+  const currentChannel = getCurrentChannel();
+  const [provider, providerId] = channel.split(':');
```

Comment:
downstream socket code really shouldn't be aware of the formation of channel strings. is there a way to abstract this out?